src/apack/zcl_abapgit_apack_reader.clas.abap

```diff
@@ -0,0 +1,101 @@
+CLASS zcl_abapgit_apack_reader DEFINITION
+  PUBLIC
+  FINAL
+  CREATE PRIVATE.
+
+  PUBLIC SECTION.
+
+    TYPES: ty_package_name TYPE devclass.
+
+    CLASS-METHODS: create_instance IMPORTING iv_package_name           TYPE ty_package_name
+                                   RETURNING VALUE(ro_manifest_reader) TYPE REF TO zcl_abapgit_apack_reader.
+    METHODS:
+      get_manifest_descriptor RETURNING VALUE(rs_manifest_descriptor)
+                                        TYPE zif_abapgit_definitions=>ty_apack_descriptor,
+      set_manifest_descriptor IMPORTING is_manifest_descriptor  TYPE zif_abapgit_definitions=>ty_apack_descriptor,
+      has_manifest RETURNING VALUE(rv_has_manifest) TYPE abap_bool.
+
+  PROTECTED SECTION.
+  PRIVATE SECTION.
+
+    TYPES: BEGIN OF ty_s_manifest_declaration,
+             clsname  TYPE seometarel-clsname,
+             devclass TYPE devclass,
+           END OF ty_s_manifest_declaration.
+
+    DATA: mv_package_name      TYPE ty_package_name,
+          ms_cached_descriptor TYPE zif_abapgit_definitions=>ty_apack_descriptor,
+          mv_is_cached         TYPE abap_bool.
+
+    METHODS: constructor IMPORTING iv_package_name TYPE ty_package_name.
+ENDCLASS.
+
+
+
+CLASS zcl_abapgit_apack_reader IMPLEMENTATION.
+
+
+  METHOD constructor.
+    me->mv_package_name = iv_package_name.
+  ENDMETHOD.
+
+
+  METHOD create_instance.
+    CREATE OBJECT ro_manifest_reader EXPORTING iv_package_name = iv_package_name.
+  ENDMETHOD.
+
+  METHOD get_manifest_descriptor.
+
```

Comment:
Sometimes, you start right behind method, and sometimes you add an extra empty line.

Reply:
Well, that might be correct. If there is any hard requirement to do either this or that, please let me know. I follow straight - as I did with this weird Hungarian notation for variables... ;)